src/coreclr/nativeaot/Runtime/wasm/FinalizerHelpers.SingleThreaded.cpp

```diff
@@ -15,7 +15,7 @@
 #include "thread.inl"
 
 // Finalizer method implemented by the managed runtime.
-extern "C" __cdecl void RhpProcessFinalizersAndReturn();
+extern "C" QCALLTYPE void RhpProcessFinalizersAndReturn();
```

Comment:
```suggestion
extern "C" void RhpProcessFinalizersAndReturn();
```
Not a QCall - QCall are methods implemented in native code that run in preemptive mode.